src/platform/mod.rs

```diff
@@ -7,20 +7,37 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-mod os {
-    #[cfg(all(not(feature = "force-inprocess"), target_os = "linux"))]
-    include!("linux/mod.rs");
+#[cfg(all(not(feature = "force-inprocess"), target_os = "linux"))]
+mod linux;
 
-    #[cfg(all(not(feature = "force-inprocess"), target_os = "macos"))]
-    include!("macos/mod.rs");
+#[cfg(all(not(feature = "force-inprocess"), target_os = "macos"))]
+mod macos;
 
-    #[cfg(any(feature = "force-inprocess", target_os = "windows", target_os = "android"))]
-    include!("inprocess/mod.rs");
-}
+#[cfg(all(not(feature = "force-inprocess"), target_os = "windows"))]
+mod windows;
 
-pub use self::os::{OsIpcChannel, OsIpcOneShotServer, OsIpcReceiver, OsIpcReceiverSet};
-pub use self::os::{OsIpcSelectionResult, OsIpcSender, OsIpcSharedMemory};
-pub use self::os::{OsOpaqueIpcChannel, channel};
+#[cfg(any(feature = "force-inprocess", target_os = "android"))]
+mod inprocess;
+
+#[cfg(all(not(feature = "force-inprocess"), target_os = "linux"))]
+pub use self::linux::{OsIpcChannel, OsIpcOneShotServer, OsIpcReceiver,
+                      OsIpcReceiverSet, OsIpcSelectionResult, OsIpcSender,
+                      OsIpcSharedMemory, OsOpaqueIpcChannel, channel};
+
+#[cfg(all(not(feature = "force-inprocess"), target_os = "macos"))]
+pub use self::macos::{OsIpcChannel, OsIpcOneShotServer, OsIpcReceiver,
+                      OsIpcReceiverSet, OsIpcSelectionResult, OsIpcSender,
+                      OsIpcSharedMemory, OsOpaqueIpcChannel, channel};
+
+#[cfg(all(not(feature = "force-inprocess"), target_os = "windows"))]
+pub use self::windows::{OsIpcChannel, OsIpcOneShotServer, OsIpcReceiver,
+                        OsIpcReceiverSet, OsIpcSelectionResult, OsIpcSender,
+                        OsIpcSharedMemory, OsOpaqueIpcChannel, channel};
+
+#[cfg(any(feature = "force-inprocess", target_os = "android"))]
+pub use self::inprocess::{OsIpcChannel, OsIpcOneShotServer, OsIpcReceiver,
+                          OsIpcReceiverSet, OsIpcSelectionResult, OsIpcSender,
+                          OsIpcSharedMemory, OsOpaqueIpcChannel, channel};
```

Comment:
Ugh, I see...

Why don't you just do `pub use self::inprocess::*` etc. though?

Also, this should be a separate PR I'd say?...